src/python/CRABClient/ClientUtilities.py

```diff
@@ -232,20 +236,22 @@ def uploadlogfile(logger, proxyfilename, logfilename=None, logpath=None, instanc
         # Encode in ascii because old pycurl present in old CMSSW versions
         # doesn't support unicode.
         cacheurl = cacheurl['cacheSSL'].encode('ascii')
-        cacheurldict = {'endpoint': cacheurl, "pycurl": True}
 
-        ufc = UserFileCache(cacheurldict)
-        logger.debug("cacheURL: %s\nLog file name: %s" % (cacheurl, logfilename))
         logger.info("Uploading log file...")
-        ufc.uploadLog(logpath, logfilename)
-        logger.info("%sSuccess%s: Log file uploaded successfully." % (colors.GREEN, colors.NORMAL))
-        logfileurl = cacheurl + '/logfile?name='+str(logfilename)
-        if not username:
-            from CRABClient.UserUtilities import getUsername
-            username = getUsername(proxyFile=proxyfilename, logger=logger)
-        logfileurl += '&username='+str(username)
+        if 'S3' in cacheurl.upper():
+            objecttype = 'clientlog'
+            uploadToS3(crabserver=crabserver, filepath=logpath, objecttype=objecttype, taskname=taskname, logger=logger)
+            logfileurl = getDownloadUrlFromS3(crabserver=crabserver, filepath=logpath, objecttype=objecttype, taskname=taskname, username=username, logger=logger)
```

Comment:
@ddain you should not need to send username nor filepath here. If it did not work w/o either it means there's something to fix.
The crabserver REST will get the username from the CMSWEB FrontEnd.
So you should not need the call to getUserName which you inserted above.
Please try and let me know if you have problems.

Reply:
you are right, it works. thank you!